src/Orleans.Reminders.CosmosDB/CosmosDBReminderTable.cs

```diff
@@ -278,7 +279,7 @@ private async Task TryCreateCosmosDBResources()
             {
                 Id = this._options.Collection
             };
-
+            remindersCollection.PartitionKey.Paths.Add(PARTITION_KEY);
```

Comment:
Just doing that is not goind to work.

```
"   at Microsoft.Azure.Documents.Client.DocumentClient.AddPartitionKeyInformationAsync(DocumentServiceRequest request, RequestOptions options)\n   at Microsoft.Azure.Documents.Client.DocumentClient.ExecuteStoredProcedurePrivateAsync[TValue](String storedProcedureLink, RequestOptions options, Object[] procedureParams)\n   at Microsoft.Azure.Documents.BackoffRetryUtility`1.<>c__DisplayClass1_0.<<ExecuteAsync>b__0>d.MoveNext()\n--- End of stack trace from previous location where exception was thrown ---\n   at Microsoft.Azure.Documents.BackoffRetryUtility`1.ExecuteRetry(Func`1 callbackMethod, Func`3 callShouldRetry, CancellationToken cancellationToken, Action`1 preRetryCallback)\n   at Microsoft.Azure.Documents.BackoffRetryUtility`1.ExecuteRetry(Func`1 callbackMethod, Func`3 callShouldRetry, CancellationToken cancellationToken, Action`1 preRetryCallback)\n   at Microsoft.Azure.Documents.BackoffRetryUtility`1.ExecuteAsync(Func`1 callbackMethod, IRetryPolicy retryPolicy, CancellationToken cancellationToken, Action`1 preRetryCallback)\n   at Orleans.Reminders.CosmosDB.CosmosDBReminderTable.ExecuteWithRetries[TResult](Func`1 clientFunc) in /Users/guto/dev/repos/OrleansContrib/Orleans.CosmosDB/src/Orleans.Reminders.CosmosDB/CosmosDBReminderTable.cs:line 234\n   at Orleans.Reminders.CosmosDB.CosmosDBReminderTable.ReadRows(UInt32 begin, UInt32 end) in /Users/guto/dev/repos/OrleansContrib/Orleans.CosmosDB/src/Orleans.Reminders.CosmosDB/CosmosDBReminderTable.cs:line 150"
```

All the tests fail. Did you ran them?

Also, can you elaborate on why `GrainId` should be a partition key? Please check the other providers to come out with something that will not affect performance.

The CI isn't enabled on this repo yet, so please make sure you `dotnet test` before submit the PR.